libraries/AP_NavEKF3/AP_NavEKF3_core.h

```diff
@@ -797,7 +810,7 @@ class NavEKF3_core : public NavEKF_core_common
     void InitialiseVariablesMag();
 
     // reset the horizontal position states uing the last GPS measurement
-    void ResetPosition(void);
+    void ResetPosition(resetDataSource posResetSource = DEFAULT);
```

Comment:
Do we really need to use `DEFAULT` here?  There aren't that many callers to change to pass through `DEFAULT`.  I have an aversion to parameter defaults :-)

Note also that adding that `enum class` would make it clear what you're passing in in the call, too:

```ResetPosition(DEFAULT);```

Becomes

```
ResetPosition(PositionResetDataSource::DEFAULT)
```

Reply:
> I was thinking this would be nicer as an enum class earlier today.
 I have an aversion to parameter defaults :-)

OK. Both are done